### DISCUSSION TITLE
Interrupt async call: Is it not supported in cpp sdk?

### DISCUSSION THREAD

**Discussion by ykkzhou** [aws/aws-sdk-cpp#2996] [Q&A] [answered]
Hi,

I'm using aws cpp sdk to do some async call, specifically sqs->receiveMessagesAsync. However, I can't find a way to interrupt the call in the middle. 

I looked up for documentation that quote ["However, the client object must continue to exist until the asynchronous operation finishes"](https://docs.aws.amazon.com/zh_cn/sdk-for-cpp/v1/developer-guide/async-methods.html) insdk-for-cpp. I want something that looks like the function [checkInterrupted](https://github.com/aws/aws-sdk-java/blob/7bf0fbec42de8a7d54875510f27fb363ca2d19f5/aws-java-sdk-core/src/main/java/com/amazonaws/http/AmazonHttpClient.java#L877) in java sdk. So, is such interruption just not supported for cpp sdk?

**Comment by jmklix** [comment, marked as answer]
Correct, there are no interrupts for this in the C++ sdk. Interrupts in C++ are generally bad and cause undefined behavior. 

You can try to abort the call in in the middle by using ["ContinueRequestHandler"](https://github.com/aws/aws-sdk-cpp/blob/67f1aae9c821f8ed061a89423741a16a4723d018/src/aws-cpp-sdk-core/include/aws/core/AmazonWebServiceRequest.h#L133). [Here](https://github.com/aws/aws-sdk-cpp/blob/67f1aae9c821f8ed061a89423741a16a4723d018/tests/aws-cpp-sdk-logs-integration-tests/CloudWatchLogsTests.cpp#L201-L211) is an example that being used in one of our tests.
